pkg/registry/core/pod/strategy.go

```diff
@@ -975,3 +977,12 @@ func apparmorFieldForAnnotation(annotation string) *api.AppArmorProfile {
 	// length or if the annotation has an unrecognized value
 	return nil
 }
+
+// setGenerationForPodSpecUpdate bumps metadata.generation if needed for any updates
+// to the podspec.
+func setGenerationForPodSpecUpdate(newPod, oldPod *api.Pod) {
+	// Spec updates bump the generation.
+	if newPod.Generation == 0 || !apiequality.Semantic.DeepEqual(newPod.Spec, oldPod.Spec) {
```

Comment:
I think the `newPod.Generation == 0` case needs to be handled differently. Nothing else is allowed to modify generation, right? So can we just unconditionally copy it from oldpod?
```suggestion
	newPod.Generation = oldPod.Generation
	if !apiequality.Semantic.DeepEqual(newPod.Spec, oldPod.Spec) {
```

Reply:
correct, newPod.Generation is generically stomped with oldPod.Generation here:

https://github.com/kubernetes/kubernetes/blob/821bc2e15e3abadb996d477f2bfb59466cec01e0/staging/src/k8s.io/apiserver/pkg/registry/rest/update.go#L122-L127

Reply:
The `newPod.Generation == 0` was more for something else that upon thinking about it more I actually don't think it's necessary. So took it out. Since newPod.Generation is stomped with oldPod.Generation before getting here, this is now just the spec equality check